libraries/AP_NavEKF3/AP_NavEKF3_core.cpp

```diff
@@ -1067,209 +1084,376 @@ void NavEKF3_core::CovariancePrediction()
     dvx_b = stateStruct.accel_bias.x;
     dvy_b = stateStruct.accel_bias.y;
     dvz_b = stateStruct.accel_bias.z;
-    float _gyrNoise = constrain_float(frontend->_gyrNoise, 0.0f, 1.0f);
-    daxVar = dayVar = dazVar = sq(dt*_gyrNoise);
+
+    bool quatCovResetOnly = false;
+    if (rotVarVecPtr != nullptr) {
+        // Handle special case where we are initialising the quaternion covariances using an earth frame
+        // vector defining the variance of the angular alignment uncertainty. Convert he varaince vector
+        // to a matrix and rotate into body frame. Use the exisiting gyro error propagation mechanism to
+        // propagate the body frame angular uncertainty variances.
+        Vector3f rotVarVec = *rotVarVecPtr;
```

Comment:
we can use a bit less stack by making this a const ref:
 const Vector3f &rotVarVec = *rotVarVecPtr;

Reply:
Done